Neo4j.Driver/Neo4j.Driver/Internal/Extensions/CollectionExtensions.cs

```diff
@@ -86,8 +88,46 @@ public static string ValueToString(this object o)
 
         public static IDictionary<string, object> ToDictionary(this object o)
         {
-            return o.GetType().GetRuntimeProperties()
-                .ToDictionary(prop => prop.Name, prop => prop.GetValue(o, null));
+            if (o == null)
+            {
+                return null;
+            }
+
+            return FillDictionary(o, new Dictionary<string, object>());
+        }
+
+        private static IDictionary<string, object> FillDictionary(object o, IDictionary<string, object> dict)
+        {
+            foreach (var propInfo in o.GetType().GetRuntimeProperties())
+            {
+                var name = propInfo.Name;
+                var value = propInfo.GetValue(o);
+
+                if (value != null)
+                {
+                    var convertible = value as IConvertible;
+                    if (convertible != null)
```

Comment:
Ideally it should be a recursive call, have a look at [java driver's impl](https://github.com/neo4j/neo4j-java-driver/blob/1.5/driver/src/main/java/org/neo4j/driver/v1/Values.java#L68)

Each elements of list is a recursive call [here](https://github.com/neo4j/neo4j-java-driver/blob/1.5/driver/src/main/java/org/neo4j/driver/v1/Values.java#L195)

And according to [this](https://github.com/neo4j/neo4j-dotnet-driver/blob/1.5/Neo4j.Driver/Neo4j.Driver/Internal/IO/PackStreamWriter.cs#L40), we have to convert array to IList too